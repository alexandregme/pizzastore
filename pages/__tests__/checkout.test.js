import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import CheckoutPage from '../checkout';
import App from '../../components/app';
import Checkout from "../../components/checkout";

describe("Checkout page", ()=>{
  let mountedCheckoutPage;

  beforeEach(()=>{
    mountedCheckoutPage = shallow(<CheckoutPage />);
  });

  it('renders without crashing', () => {
    shallow(<CheckoutPage />);
    expect(mountedCheckoutPage).toMatchSnapshot();
  });

  it('expect to have App component', () => {
    expect(mountedCheckoutPage.find(App)).toHaveLength(1);
  });

  it('expect to have Checkout component', () => {
    expect(mountedCheckoutPage.find(Checkout)).toHaveLength(1);
  });
});
