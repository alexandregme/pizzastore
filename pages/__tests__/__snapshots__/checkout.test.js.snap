// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout page renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckoutPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Checkout />,
      "title": "Checkout",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Checkout />,
        "title": "Checkout",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
